### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by zorglube**
- Handle AutoCloseable Connexion and Database
 - Improved Resource loading

### CODE REVIEW

**Comment by zorglube** [comment]
@lbruun : here is a king of rebase ;-)

**Comment by zorglube** [comment]
Hey @lbruun, 

Yes IDE certainly have a formatter different than your's. I'll handle the formatting issue. 

About the formatting, I'll make some updates on the PR about Formatting I have made previously. 

See you ;-)

**Comment by zorglube** [comment]
@lbruun : I handled whitespaceschanges.

File: autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java [hunk 1]
```diff
@@ -15,30 +15,43 @@
  */
 package net.lbruun.springboot.preliquibase;
 
-import jakarta.validation.constraints.NotEmpty;
-import org.springframework.boot.context.properties.ConfigurationProperties;
-import org.springframework.validation.annotation.Validated;
+import static java.lang.String.format;
```
Comment (lbruun) [inline autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java:18]: Not a fan of these static imports. It makes the code _less_ readable. 

In my mind

`LiquibaseUtils.getLiquibaseDatabaseShortName(...)` 

is a lot easier to read than

`getLiquibaseDatabaseShortName(...)` 

because the former makes it obvious that the method is something from outside the class.

File: autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java [hunk 2]
```diff
@@ -15,30 +15,43 @@
...
+import net.lbruun.springboot.preliquibase.PreLiquibaseException.SqlScriptRefError;
 
 /**
  * Properties for Pre-Liquibase module.
  *
  * @author lbruun
  */
-@Validated
-@ConfigurationProperties(prefix = PROPERTIES_PREFIX)
+@ConfigurationProperties(prefix = "preliquibase")
```
Comment (lbruun) [inline autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java:49]: Why this change?

File: autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java [hunk 3]
```diff
@@ -221,7 +259,41 @@ public List<String> getSqlScriptReferences() {
      *
      * @param sqlScriptReferences list of Spring Resource references.
      */
-    public void setSqlScriptReferences(List<String> sqlScriptReferences) {
+    // @formatter:on
```
Comment (lbruun) [inline autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java:262]: What is this?  No IDE specific stuff in the code, please.

File: pom.xml [hunk 4]
```diff
@@ -87,10 +85,14 @@ limitations under the License.
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <dependency>
```
Comment (lbruun) [inline pom.xml:88]: I see the point here. Don't really know if it improves anything.

In any case the elements are now misaligned and for this reason alone I'll have to reject it (regardless if the idea is a good one or not .. which I cannot quite decide upon yet)

File: autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibase.java [hunk 5]
```diff
@@ -5,7 +5,7 @@
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *      http://www.apache.org/licenses/LICENSE-2.0
```
Comment (lbruun) [inline autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibase.java:8]: Ooops. Why ????
This is the text from the AL2 license. It must not be touched by an (accidental) re-format of code.

**Comment by lbruun** [comment]
@zorglube . You are still re-formatting all of the code to the way you have set _your_ formatter. Don't touch formatting when contributing to FOSS projects. Just look at the convention used and go with that. 
 
Also, I have a bunch of other comments on things I do not like in the PR. I've commented directly in a few places. 
 
I don't want this to be too discouraging for you, but you are simply attempting to change too many things in a single PR. It therefore becomes impossible to review.  I'm not able to "see the wood for the trees", so to speak. (l'arbre cache la forêt).

For this reason, I'll have to reject. 

Your idea of taking advantage of the fact that the Liquibase [`Database`](https://javadoc.io/static/org.liquibase/liquibase-core/4.17.2/liquibase/database/Database.html) and [`DatabaseConnection`](https://javadoc.io/static/org.liquibase/liquibase-core/4.17.2/liquibase/database/DatabaseConnection.html) now implements `Autocloseable` - and that the code in the project's `LiquibaseUtils` class can therefore now be greatly simplified - is a good one and I'll implement that and credit you. Thanks.

Your other idea, changing the type of `sqlScriptReferences` from `List<String>` to `List<Resource>`  in the `PreLiquibaseProperties` class, is still something I'm not convinced about. I would have to see it in a change which _only_ does that in order to evaluate if it is a simplification of the code or not, or brings other benefits.

**Comment by lbruun** [comment]
See PR #20

**Comment by zorglube** [comment]
> Your other idea, changing the type of `sqlScriptReferences` from `List<String>` to `List<Resource>` in the `PreLiquibaseProperties` class, is still something I'm not convinced about. I would have to see it in a change which _only_ does that in order to evaluate if it is a simplification of the code or not, or brings other benefits.

It simplify a lot the resource loading. Using Spring `Resource` allow you to provide file, http, classpath and so on file URI without having to handle the the difference by hand. 

I.e.: Using Resource allow to have `script = classpath:/foo/bar/file.sql` or `script = file:/foo/bar/file.sql` even `script = http://uri.tld/foo/bar/file.sql`. 

Like it's written in the Spring doc : https://docs.spring.io/spring-framework/docs/6.0.x/reference/html/core.html#resources

**Comment by zorglube** [comment]
> @zorglube . You are still re-formatting all of the code to the way you have set _your_ formatter. Don't touch formatting when contributing to FOSS projects. Just look at the convention used and go with that.

@lbruun I purposed a PR that contain an Eclipse formatter for the project, if might not be perfect today. 
If you have a formatter, or if you can extract your formatter configuration, it can help to end the related problems.

**Comment by zorglube** [comment]
> I don't want this to be too discouraging for you, but you are simply attempting to change too many things in a single PR. It therefore becomes impossible to review. I'm not able to "see the wood for the trees", so to speak. (l'arbre cache la forêt).

I understand, I'll have a look and maybe purpose some other PR.

File: autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java (hunk 1 above)
Comment (zorglube) [reply autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java:18]: It's a matter of personal habits and preferences. Anyhow since it's your project, I'll comply.

File: autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java (hunk 2 above)
Comment (zorglube) [reply autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java:49]: Because after my refactoring it has become useless to have the configuration key prefix stored into a static var.

File: autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java (hunk 3 above)
Comment (zorglube) [reply autoconfigure/src/main/java/net/lbruun/springboot/preliquibase/PreLiquibaseProperties.java:262]: It's an Eclipse specific comment. Like I written before, your project your rules.

**Comment by lbruun** [comment]
> > Your other idea, changing the type of `sqlScriptReferences` from `List<String>` to `List<Resource>` in the `PreLiquibaseProperties` class, is still something I'm not convinced about. I would have to see it in a change which _only_ does that in order to evaluate if it is a simplification of the code or not, or brings other benefits.
> 
> @lbruun Using the Spring Resource simplify a lot the resource loading. Using Spring `Resource` allow you to provide file, http, classpath and so on file URI without having to handle the the difference by hand. Using Resource allow to have `script = classpath:/foo/bar/file.sql` or `script = file:/foo/bar/file.sql` even `script = http://uri.tld/foo/bar/file.sql`.
> 
> Like it's written in the Spring doc : https://docs.spring.io/spring-framework/docs/6.0.x/reference/html/core.html#resources
> 

But this is already the case. It is just initially represented as a String. All resource abstractions that Spring supports are supported, by definition. None of these Spring Resource abstractions are handled "by hand". 

As indicated the code originally comes from Spring 2.3, class [`DataSourceInitializer`](https://github.com/spring-projects/spring-boot/blob/v2.3.1.RELEASE/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceInitializer.java), so at least some point in time someone from Spring Boot project decided that this was a good way to do this. This is not to say that the Spring folks are always right. I can also see that they've reorganized later but it looks to me as if they are are still using `List<String>` even in Spring Boot 3.0.
